src/xbot/common/command/BaseRobot.java

```diff
@@ -188,7 +196,12 @@ else if(timeSinceLastLog >= frequencyReportInterval.get()) {
             lastFreqCounterResetTime = Timer.getFPGATimestamp();
             
             log.info("Robot loops per second: " + loopsPerSecond);
+            
+            log.info("Total Current: " + pdp.getTotalCurrent());
+            
+            log.info("Loop Duration:" + loopDuration);
```

Comment:
So there's still a logic problem here - you only report the most recent loop duration. After all, this chunk of code is only called every 10 seconds or so, during which there should have been several hundred loops.

What you need to do, after calculating the loop duration above, is accumulate it somehow. Then, in this code, you take the accumulated value and divide it by the number of loops, thus measuring the "average loop duration."